plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseHiveConnectorTest.java

```diff
@@ -4085,6 +4089,55 @@ public void testWritersAcrossMultipleWorkersWhenScaleWritersIsEnabled()
                 .isBetween((long) taskMaxScaleWriterCount + workers, workers * taskMaxScaleWriterCount);
     }
 
+    @DataProvider(name = "taskWritersLimitParams")
+    public Object[][] prepareScaledWritersOption()
+    {
+        return new Object[][] {{true, true, 2}, {false, true, 2}, {false, false, 3}};
+    }
+
+    @Test(dataProvider = "taskWritersLimitParams")
+    public void testWriterTasksCountLimitUnpartitioned(boolean scaleWriters, boolean redistributeWrites, int expectedFilesCount)
+    {
+        testLimitWriterTasks(2, expectedFilesCount, scaleWriters, redistributeWrites, false);
+    }
+
+    @Test
+    public void testWriterTasksCountLimitPartitionedScaleWritersDisabled()
+    {
+        testLimitWriterTasks(2, 2, false, true, true);
+    }
+
+    @Test
+    public void testWriterTasksCountLimitPartitionedScaleWritersEnabled()
```

Comment:
Maybe no need to test this since `scaleWriters` doesn't really work for partitioned data. So basically above test is equivalent to this.

Reply:
I know that `scaleWriters` actually does not work for partitioned data. Anyway, this is a path in the code to support it (in `AddExchanges`) and this is why I am testing it.